### DISCUSSION TITLE
Rename form actions to submissions (or submits)

### DISCUSSION THREAD

**Discussion by ioucyf** [sveltejs/kit#6903] [Ideas]
And let actions refer exclusively to the directive on elements.

**Comment by dummdidumm** [comment]
We thought about naming it something similar, but the resulting method names where all too clunky, slightly semantically incorrect, or too verbose.
- `export let submissions = {..}` is not 100% semantically correct. It's acting on a submission, not the submission itself
- `applySubmission` is wrong, `applySubmissionResult` is too long

**Reply by vekunz** [reply]
The problem with the current name "actions" is that it sounds like general actions. But in fact, there are not really general actions but only form actions. I just noticed that actions do not allow sending data other than form data (for example JSON is not allowed), but not every action on a website is suitable to be a &lt;form>. So if it is not planned to expand the usage of actions, they should somehow be renamed, so that it is clear that they are only for &lt;form> and for nothing else.

**Reply by ZetiMente** [reply]
why not formActions ?
